src/catalog/src/durable/persist.rs

```diff
@@ -603,17 +617,18 @@ impl<T: TryIntoStateUpdateKind, U: ApplyUpdate<T>> PersistHandle<T, U> {
                             .map(Into::<StateUpdate<StateUpdateKindJson>>::into)
                             .map(|update| {
                                 let kind = T::try_from(update.kind).expect("kind decoding error");
```

Comment:
When I encountered this issue the panic was coming from here, when we try to convert a `StateUpdateKindJson` into a `StateUpdateKind`. So I think we need to just move the `BTreeMap` grouping down into this branch?

Reply:
Just pushed an update.